execution_engine/src/storage/store/mod.rs

```diff
@@ -5,30 +5,36 @@ pub(crate) mod tests;
 use casper_types::bytesrepr::{self, Bytes, FromBytes, ToBytes};
 
 pub use self::store_ext::StoreExt;
-use crate::storage::transaction_source::{Readable, Writable};
 
-/// Store is responsible for abstracting `get` and `put` operations over the underlying store
-/// specified by its associated `Handle` type.
-pub trait Store<K, V> {
-    /// Errors possible from this store.
+/// Base trait for db operations that can raise an error.
+pub trait ErrorSource: Sized {
+    /// Type of error held by the trait.
     type Error: From<bytesrepr::Error>;
+}
 
-    /// Underlying store type.
-    type Handle;
+/// Represents a store that can be read from.
+pub trait Readable: ErrorSource {
```

Comment:
Why did you change from type member to inheritence?

Reply:
This was needed to handle everywhere that Transaction's Self::Error type was referenced. I tried adding Error as an associated type but it wasn't as compatible with our existing top level functions (`write`, `put_stored_values` etc) trait bounds, so I settled on this approach.